# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../openoffice-org/patches/libxmlsec-findcerts.diff
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- libxmlsec/xmlsec1-1.2.6.patch
+++ libxmlsec/xmlsec1-1.2.6.patch
@@ -17395,3 +17395,25 @@
   XMLSEC_NSS_ALIBS    	= smime3.lib ssl3.lib nss3.lib libnspr4_s.lib libplds4_s.lib libplc4_s.lib kernel32.lib user32.lib gdi32.lib
   
   XMLSEC_MSCRYPTO_SOLIBS  = kernel32.lib user32.lib gdi32.lib Crypt32.lib Advapi32.lib
+*** misc/xmlsec1-1.2.6/src/nss/x509.c	2007-03-16 12:51:38.000000000 -0400
+--- misc/build/xmlsec1-1.2.6/src/nss/x509.c	2007-03-19 11:23:22.000000000 -0400
+*************** xmlSecNssKeyDataX509VerifyAndExtractKey(
+*** 1365,1373 ****
+  		/*-
+  		 * I'll search key according to KeyReq.
+  		 */
+-         slot = cert->slot ;
+          if( ( keyInfoCtx->keyReq.keyType & xmlSecKeyDataTypePrivate ) == xmlSecKeyDataTypePrivate ) {
+!             if( ( priKey = PK11_FindPrivateKeyFromCert( slot , cert , NULL ) ) == NULL ) {
+                  xmlSecError( XMLSEC_ERRORS_HERE ,
+                      xmlSecErrorsSafeString( xmlSecKeyDataGetName( data ) ) ,
+                      "PK11_FindPrivateKeyFromCert" ,
+--- 1365,1372 ----
+  		/*-
+  		 * I'll search key according to KeyReq.
+  		 */
+          if( ( keyInfoCtx->keyReq.keyType & xmlSecKeyDataTypePrivate ) == xmlSecKeyDataTypePrivate ) {
+!             if( ( priKey = PK11_FindKeyByAnyCert( cert , NULL ) ) == NULL ) {
+                  xmlSecError( XMLSEC_ERRORS_HERE ,
+                      xmlSecErrorsSafeString( xmlSecKeyDataGetName( data ) ) ,
+                      "PK11_FindPrivateKeyFromCert" ,
